pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -829,3 +831,9 @@ def is_distributed(self) -> bool:
         if isinstance(self.accelerator, TPUAccelerator):
             is_distributed |= self.strategy.is_distributed
         return is_distributed
+
+
+def _populate_registries() -> None:
```

Comment:
Why have it in `accelerator_connector`, not during import time `__init__`?

Reply:
Why not? What's the benefit of delaying it? 

If I do this 

```python
    from pytorch_lightning.accelerators.registry import ACCELERATOR_REGISTRY
    print(ACCELERATOR_REGISTRY.names)
```

I would be expecting a list of accelerator names, rather than calling one more function `_populate_registeries` before it. Also `registeries` is plural, and we are populating only one Registry

Reply:
Doing things at import time is usually problematic. In addition to slowing time to start running, it means users are very limited when they want to override or customize behaviour.

> Also registeries is plural, and we are populating only one Registry

Yes, but we would eventually do the same changes for strategies.

Reply:
For strategies, we really can't get rid of `register_strategies`, as we have multiple configurations for certain strategies.
That's the major con of this approach, we can't have multiple subclasses with different defaults in our code.

We can't have an inconsistent api for registering strategies and accelerators.